subsys/net/ip/tcp.c

```diff
@@ -2358,6 +2358,16 @@ NET_CONN_CB(tcp_syn_rcvd)
 		return NET_DROP;
 	}
 
+#if defined(CONFIG_NET_IPV4)
```

Comment:
How is this related to #11329, as stated in the commit message?

Reply:
ipv4_arp_multicast_icmp.pcap.gz has this packet too. TCP SYN packet with multicast src address.

Reply:
Ok, but that's not what #11329 is talking about, there it's only about arp table that should not be updated etc... 

If that syn/mcast is a bug, it's a different one. So do not set Fixes #11329